### PR TITLE
:sparkles: Add in catalog label selection support

### LINKED ISSUE #1112: Hook label selection into resolution + test

### PR DESCRIPTION
**Pull request by tmshort**
Fix #1112

Adds some tests to validate the label selector.
Modified some tests to add labels and selectors. They are not necessary now, but will be once 1110 is resolved; as that issue adds the ambiguity that this PR helps resolve.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Comment by tmshort** [comment]
Adding a reference to #1110 here, so it doesn't get resolved when this is merged...

File: internal/resolve/catalog_test.go [hunk 1]
```diff
@@ -589,3 +589,65 @@ func genPackage(pkg string) *declcfg.DeclarativeConfig {
...
+		},
+		Spec: ocv1alpha1.ClusterExtensionSpec{
+			PackageName: "foo",
+			CatalogSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"": "value"},
+			},
+		},
+	}
+	_, _, _, err := r.Resolve(context.Background(), ce, nil)
+	assert.EqualError(t, err, "desired catalog selector is invalid: key: Invalid value: \"\": name part must be non-empty; name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')")
```
Comment (dtfranz) [inline internal/resolve/catalog_test.go:633]: Nit: I think just `wantErr: true` or `assert.ErrorContains(t, err, "selector is invalid")` is probably sufficient.

File: internal/resolve/catalog_test.go [hunk 2]
```diff
@@ -589,3 +589,65 @@ func genPackage(pkg string) *declcfg.DeclarativeConfig {
 		Deprecations: []declcfg.Deprecation{packageDeprecation(pkg)},
 	}
 }
+
```
Comment (dtfranz) [inline internal/resolve/catalog_test.go:592]: Could you also add a positive case here? i.e. given multiple catalogs and valid label selector, one catalog is chosen.

File: internal/resolve/catalog_test.go (hunk 1 above)
Comment (tmshort) [reply internal/resolve/catalog_test.go:633]: Yeah, I copied pasted...

File: internal/resolve/catalog_test.go (hunk 2 above)
Comment (tmshort) [reply internal/resolve/catalog_test.go:605]: I had to add in support for labels to the catalog walker, but I think I got it.

**Pull request merged**